sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SparkSQLParser.scala

```diff
@@ -66,7 +66,13 @@ class SqlLexical(val keywords: Seq[String]) extends StdLexical {
 
   override lazy val token: Parser[Token] =
     ( identChar ~ (identChar | digit).* ^^
-      { case first ~ rest => processIdent((first :: rest).mkString) }
+      { case first ~ rest => 
+          val rsIdent = processIdent((first :: rest).mkString.toLowerCase)
```

Comment:
`toLowerCase` probably causes some other issue, can you add a unit test for this?